dmriprep/workflows/dwi/util.py

```diff
@@ -292,3 +292,181 @@ def _normalize(in_file, newmax=2000, perc=98.0):
     hdr.set_data_dtype("int16")
     nii.__class__(data.astype("int16"), nii.affine, hdr).to_filename(out_file)
     return out_file
+
+
+def gen_index(in_file):
+    # Generate the index file for eddy
+    import os
+    import numpy as np
+    import nibabel as nib
+    from nipype.utils.filemanip import fname_presuffix
+
+    out_file = fname_presuffix(
+        in_file,
+        suffix="_index.txt",
+        newpath=os.path.abspath("."),
+        use_ext=False,
+    )
+    vols = nib.load(in_file).shape[-1]
+    index_lines = np.ones((vols,))
+    index_lines_reshape = index_lines.reshape(1, index_lines.shape[0])
+    np.savetxt(out_file, index_lines_reshape, fmt="%i")
+    return out_file
+
+
+def gen_acqparams(in_file, total_readout_time=0.05):
```

Comment:
```suggestion
def gen_acqparams(in_file, total_readout_time, pe_dir):
```
Better fail rather than using a default value